ansible/library/shell_cmds.py

```diff
@@ -76,19 +77,25 @@
         - ls /home
         - pwd
     continue_on_fail: False
+    timeout: 30
 '''
 
 
-def run_cmd(module, cmd):
-
-    rc, out, err = module.run_command(cmd, use_unsafe_shell=True)
+def run_cmd(module, cmd, timeout):
+    cmd_with_timeout = "echo '{}' | timeout --preserve-status {} bash".format(cmd, timeout)
```

Comment:
please move this before line 89.
And we need additional check for invalid char

Reply:
Thanks, updated the code. For the commands cannot be wrapped with `timeout`, will run it without time limit and return a warning message.